Assets/LeapMotion/Widgets/Scripts/Dial/FadeTextByRotation.cs

```diff
@@ -0,0 +1,110 @@
+﻿using UnityEngine;
+using UnityEngine.UI;
+using System.Collections;
+
+/// <summary>
+/// Fade out text label in dial as it rotates away.
+/// </summary>
+/// <remarks>
+/// Uses the dot product between the parent of the label's forward direction
+/// and the forward direction of the label, passed through a curve filter 
+/// to determine the opacity to set the text label to.
+/// </remarks>
+public class FadeTextByRotation : MonoBehaviour {
+  /// <summary>
+  /// Curve to translate dot product to opacity
+  /// </summary>
+  public AnimationCurve FadeCurve;
+
+  /// <summary>
+  /// Reference for the "forward direction.
+  /// </summary>
+  /// <remarks>
+  /// Uses the label's parent's parent (label -> dial center -> panel center)
+  /// NOTE: This might deserve an override but erring on the side of 
+  /// fewer editor hookups to have break.
+  /// 
+  /// "Forward" is assumed to be -z.
+  /// </remarks>
+  private Transform m_referenceTransform;
```

Comment:
Please make public, and attempt finding only if it turns out to be null when scene begins.